core/components/tree/treecontrol.js

```diff
@@ -0,0 +1,445 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2019 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Definition of the Blockly.tree.TreeControl class.
+ * This class is similar to Closure's goog.ui.tree.TreeControl class.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+'use strict';
+
+goog.provide('Blockly.tree.TreeControl');
+
+goog.require('Blockly.Events');
+goog.require('Blockly.Events.Ui');
+goog.require('Blockly.tree.TreeNode');
+goog.require('Blockly.tree.BaseNode');
+goog.require('Blockly.utils.aria');
+
+
+/**
+ * An extension of the TreeControl object in closure that provides
+ * a way to view a hierarchical set of data.
+ * Similar to Closure's goog.ui.tree.TreeControl
+ *
+ * @param {Blockly.Toolbox} toolbox The parent toolbox for this tree.
+ * @param {Object} config The configuration for the tree.
+ * @constructor
+ * @extends {Blockly.tree.BaseNode}
+ */
+Blockly.tree.TreeControl = function(toolbox, config) {
+  this.toolbox_ = toolbox;
+  Blockly.tree.BaseNode.call(this, '', config);
+
+  // The root is open and selected by default.
+  this.setExpandedInternal(true);
+  this.setSelectedInternal(true);
+
+  this.selectedItem_ = this;
+};
+goog.inherits(Blockly.tree.TreeControl, Blockly.tree.BaseNode);
+
+/**
+ * Returns the tree.
+ * @override
+ */
+Blockly.tree.TreeControl.prototype.getTree = function() {
+  return this;
+};
+
+/**
+ * Return node depth.
+ * @override
+ */
+Blockly.tree.TreeControl.prototype.getDepth = function() {
+  return 0;
+};
+
+/**
+ * Expands the parent chain of this node so that it is visible.
+ * @override
+ */
+Blockly.tree.TreeControl.prototype.reveal = function() {
+  // always expanded by default
+  // needs to be overriden so that we don't try to reveal our parent
+  // which is a generic component
+};
+
+
+/**
+ * Handles focus on the tree.
+ * @param {!Event} e The browser event.
+ * @private
+ */
+Blockly.tree.TreeControl.prototype.handleFocus_ = function(/* e */) {
```

Comment:
You can use `_e` to have an unused variable without commenting it out (closure compiler may complain otherwise)

Reply:
Cool, thanks.